### PR TITLE
Deprecate gubernator!

### PR DESCRIPTION
**Pull request by Katharine**
This PR changes prow and testgrid configuration to point at Prow, and adds a message to the top of Spyglass pages pointing to Gubernator.

This cannot merge before #11274, #11299, #11292 (or similar), #11207, and #11183 have been actually deployed. Most have merged, but are blocked on our ongoing inability to deploy prow.

/assign @BenTheElder @fejta @spiffxp @krzyzacy @cjwagner 
/hold

### CODE REVIEW

File: prow/config.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
...
+  report_template: '[Full PR test history](https://prow.k8s.io/view/pr-history?org={{.Spec.Refs.Org}}&repo={{.Spec.Refs.Repo}}&pr={{with index .Spec.Refs.Pulls 0}}{{.Number}}{{end}}). [Your PR dashboard](https://gubernator.k8s.io/pr/{{with index .Spec.Refs.Pulls 0}}{{.Author}}{{end}}). Please help us cut down on flakes by [linking to](https://git.k8s.io/community/contributors/devel/flaky-tests.md#filing-issues-for-flaky-tests) an [open issue](https://github.com/{{.Spec.Refs.Org}}/{{.Spec.Refs.Repo}}/issues?q=is:issue+is:open) when you hit one in your PR.'
```
Comment (oomichi) [inline prow/config.yaml]: Does "[Your PR dashboard]" still need to use gubernator?

Comment (fejta) [reply prow/config.yaml]: Yes, the job log part of gubernator is what's deprecated

**Comment by cblecker** [comment]
Will gubernator still be available, but just not be the default?

/hold
Explicit hold as this is a contributor impacting change, and needs to go through the notification/consensus process when it's ready to go.

**Comment by Katharine** [comment]
@cblecker Gubernator will still be available for the forseeable future (though presumably not indefinitely). Spyglass pages will have a banner at the top with a link pointing to the corresponding Gubernator page, something like this:

**Comment by cblecker** [comment]
@Katharine that looks great! No objections from me.

When we're ready to go with this:
- We need to send an e-mail to testing, contribex, and k-dev outlining the change to experience, what folks can expect, and where to send feedback.
- At least 72 hours notice required, more if over a weekend, for feedback/lazy consensus.
- Then :shipit: !

File: config/tests/jobs/jobs_test.go [hunk 2]
```diff
@@ -42,7 +42,7 @@ import (
 
 var configPath = flag.String("config", "../../../prow/config.yaml", "Path to prow config")
 var jobConfigPath = flag.String("job-config", "../../jobs", "Path to prow job config")
-var gubernatorPath = flag.String("gubernator-path", "https://gubernator.k8s.io", "Path to linked gubernator")
+var deckPath = flag.String("deck-path", "https://prow.k8s.io", "Path to deck")
```
Comment (fejta) [inline config/tests/jobs/jobs_test.go]: why is this a flag?

Comment (Katharine) [reply config/tests/jobs/jobs_test.go]: ¯\\\_(ツ)\_/¯

Comment (krzyzacy) [reply config/tests/jobs/jobs_test.go]: @fejta I'm kinda using this test for gerrit jobs as well so...

**Comment by Katharine** [comment]
At this point our primary blocker is the fact that we are still unable to deploy Prow. #11292 is only blocked on #11318 and a kubekins bump, and that's the last dependency. Once we finally manage to deploy prow (hopefully by Tuesday?), I intend to send roughly this email:

>Hi all!
>
>We have been working on a replacement for Gubernator, which you may know as the page that the "Details" links on GitHub point at.
>
>This replacement, called Spyglass, is now ready for prime time, and we would like to switch all the links to Gubernator to instead be links to Spyglass.
>
>From a user perspective, Spyglass should provide the functionality of Gubernator, but with a hopefully less confusing UI. However, we will keep running Gubernator for the forseeable future, with a link to the corresponding Gubernator page at the top of each Spyglass page.
>
>Spyglass is part of Prow, and enables us to avoid maintaining an extra service for job viewing. It is also designed to be extensible, and in particular open the door for artifact- and job-specific views.
>
>You can try out Spyglass today by clicking the "View in Spyglass" link at the top of Gubernator result pages. Please send general feedback to #sig-testing on Slack.
>
>Do note that this does not affect the PR dashboard at https://gubernator.k8s.io/pr, which we have no current plans to replace and will continue to be supported.
>
>In the absence of any objections, I intend to switch this over for jobs starting on Friday.
>
>Thanks!
>\- Katharine

**Comment by Katharine** [comment]
All the dependencies are in. Once we actually manage to deploy Prow (today? maybe? hopefully?) I will send that email.

**Comment by Katharine** [comment]
The email has been sent out.

**Comment by Katharine** [comment]
Nobody has objected during the lazy consensus period, so this is good to merge in principle.

Some bugs were reported and have since been fixed, but not yet deployed. In particular, until #11463 and #11466 have merged and been deployed, we probably don't want to merge this.

/hold

**Comment by spiffxp** [comment]
explicit +1

**Comment by stevekuznetsov** [comment]
@Katharine will my Deck point to My Gubernator for this change?

File: prow/config.yaml [hunk 3]
```diff
@@ -37,6 +36,7 @@ deck:
       - "buildlog"
       "artifacts/junit.*\\.xml":
       - "junit"
+    announcement: "The old job viewer, Gubernator, has been deprecated in favour of this page, Spyglass.{{if .ArtifactPath}}For now, the old page is <a href='https://gubernator.k8s.io/build/{{.ArtifactPath}}'>still available</a>.{{end}} Please send feedback to sig-testing."
```
Comment (fejta) [inline prow/config.yaml]: Steve you'll need to set this config value to have it link back to your gubernator

**Pull request merged**